microprofile/graphql/server/src/main/java/io/helidon/microprofile/graphql/server/Schema.java

```diff
@@ -0,0 +1,562 @@
+                return schemaType;
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Return a {@link SchemaInputType} that matches the type name.
+     *
+     * @param inputTypeName type name to match
+     * @return a {@link SchemaInputType} that matches the type name or null if none found
+     */
+    public SchemaInputType getInputTypeByName(String inputTypeName) {
+        for (SchemaInputType schemaInputType : listInputTypes) {
+            if (schemaInputType.name().equals(inputTypeName)) {
+                return schemaInputType;
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Return a {@link SchemaType} that matches the given class.
+     *
+     * @param clazz the class to find
+     * @return a {@link SchemaType} that matches the given class or null if none found
+     */
+    public SchemaType getTypeByClass(String clazz) {
+        for (SchemaType schemaType : listSchemaTypes) {
+            if (clazz.equals(schemaType.valueClassName())) {
+                return schemaType;
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Return a {@link SchemaEnum} that matches the enum name.
+     *
+     * @param enumName type name to match
+     * @return a {@link SchemaEnum} that matches the enum name or null if none found
+     */
+    public SchemaEnum getEnumByName(String enumName) {
+        for (SchemaEnum schemaEnum1 : listSchemaEnums) {
+            if (schemaEnum1.name().equals(enumName)) {
+                return schemaEnum1;
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Return a {@link SchemaScalar} which matches the provided class name.
+     *
+     * @param actualClazz the class name to match
+     * @return {@link SchemaScalar} or null if none found
+     */
+    public SchemaScalar getScalarByActualClass(String actualClazz) {
+        for (SchemaScalar schemaScalar : getScalars()) {
+            if (schemaScalar.actualClass().equals(actualClazz)) {
+                return schemaScalar;
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Return a {@link SchemaScalar} which matches the provided scalar name.
+     *
+     * @param scalarName the scalar name to match
+     * @return {@link SchemaScalar} or null if none found
+     */
+    public SchemaScalar getScalarByName(String scalarName) {
+        for (SchemaScalar schemaScalar : getScalars()) {
+            if (schemaScalar.name().equals(scalarName)) {
+                return schemaScalar;
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Return true of the {@link SchemaType} with the the given name is present for this {@link Schema}.
+     *
+     * @param type type name to search for
+     * @return true if the type name is contained within the type list
+     */
+    public boolean containsTypeWithName(String type) {
+        return listSchemaTypes.stream().filter(t -> t.name().equals(type)).count() == 1;
```

Comment:
`anyMatch` could save few iterations, what if there is more than one SchemaType with the same name?